type.go

```diff
@@ -167,9 +176,17 @@ func parseRequestType(t reflect.Type) (*requestType, error) {
 	if t.Kind() != reflect.Ptr || t.Elem().Kind() != reflect.Struct {
 		return nil, fmt.Errorf("type is not pointer to struct")
 	}
+
+	var reflectType reflect.Type
```

Comment:
This is unnecessary - we just checked that it's a pointer, so we can still use the t.Elem() form as before.